core/src/test/kotlin/org/jetbrains/jewel/PainterHintTest.kt

```
@@ -0,0 +1,122 @@
+
+    @Test
+    fun `empty hint should be ignored`() = runComposeTest({
+        CompositionLocalProvider(LocalIsDarkTheme provides false) {
+            val provider = rememberResourcePainterProvider("icons/github.svg", PainterHintTest::class.java)
+
+            val painter1 by provider.getPainter()
+            // must be ignored the None and hit cache
+            val painter2 by provider.getPainter(PainterHint.None)
+            // must be ignored the None and hit cache too
+            val painter3 by provider.getPainter(PainterHint.None, PainterHint.None)
+
+            Assert.assertEquals(painter1, painter2)
+            Assert.assertEquals(painter3, painter2)
+        }
+    }) {
+        awaitIdle()
+    }
+
+    private fun String.applyPathHints(vararg hints: PainterHint): String {
+        var result = this
+        hints.forEach {
+            result = when (it) {
+                is PainterResourcePathHint -> it.patch(result, emptyList())
+                is PainterPathHint -> it.patch(result)
+                else -> return@forEach
+            }
+        }
+        return result
+    }
+
+    @Test
+    fun `dark painter hint`() {
+        val basePath = "icons/github.svg"
+        basePath.applyPathHints(Dark(true)).let {
+            Assert.assertEquals("icons/github_dark.svg", it)
+        }
+    }
+
+    @Test
+    fun `override painter hint`() {
+        val basePath = "icons/github.svg"
+        basePath.applyPathHints(Override(mapOf("icons/github.svg" to "icons/search.svg"))).let {
+            Assert.assertEquals("icons/search.svg", it)
+        }
+    }
+
+    @Test
+    fun `selected painter hint`() {
+        val basePath = "icons/checkbox.svg"
+        basePath.applyPathHints(Selected(true)).let {
+            Assert.assertEquals("icons/checkboxSelected.svg", it)
+        }
+    }
+
+    @Test
+    fun `size painter hint`() {
+        val basePath = "icons/github.svg"
+        basePath.applyPathHints(Size("20x20")).let {
```

Should we also test cases where the Size gets an empty and/or blank string? I'd argue it should throw an error in that case. And for the `Int, Int` variant, we should also throw when the user passes zero or negative values, and have corresponding tests. What do you think?

---

In my impl it will be ignored when pass empty string or null to Size